whois-api/src/main/java/net/ripe/db/whois/api/rdap/domain/Redaction.java

```diff
@@ -0,0 +1,117 @@
+package net.ripe.db.whois.api.rdap.domain;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import jakarta.xml.bind.annotation.XmlAccessType;
+import jakarta.xml.bind.annotation.XmlAccessorType;
+import jakarta.xml.bind.annotation.XmlElement;
+import jakarta.xml.bind.annotation.XmlType;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "Redaction", propOrder = {
+        "name",
+        "reason",
+        "prePath",
+        "method"
+})
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class Redaction implements Serializable {
+    @XmlElement(required = true)
+    private Description name;
+
+    private Description reason;
+
+    @XmlElement(required = true)
+    private String prePath;
+
+    private String method;
+
+    public Redaction() {
+        // required no-arg constructor
+    }
+    public Redaction(final String name, final String prePath, final String reason){
+        this.name = new Description(name);
+        this.prePath = prePath;
+        this.reason = new Description(reason);
+        this.method = "removal";
+    }
+
+    public Description getName() {
+        return name;
+    }
+
+    public Description getReason() {
+        return reason;
+    }
+
+    public String getMethod() {
+        return method;
+    }
+
+    public String getPrePath() {
+        return prePath;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
```

Comment:
o can never be null here as this == o will return false anyways ? Also can you use  instanceof instaed checking via class?

Reply:
this was autogenerated by the code. So I am not sure, I will check

Reply:
Done